src/traces/surface/defaults.js

```diff
@@ -23,14 +23,18 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
     }
 
+    var x = coerce('x');
+    var y = coerce('y');
+
     var z = coerce('z');
-    if(!z) {
+    if(!z || !z.length ||
+       (x ? (x.length < 2) : false) ||
```

Comment:
I would prefer keeping things consistent with `contour` traces here. 1-item contour traces are (like 1-item surfaces) ill-defined, but still appear as `visible: true` in the `fullData` (see -> https://codepen.io/etpinard/pen/oJZyEO)

To avoid drawing errors, we should handle the 1-item case in `surface/convert.js`. Thanks!